include/boost/multiprecision/cpp_dec_float.hpp

```diff
@@ -1933,7 +1938,19 @@ bool cpp_dec_float<Digits10, ExponentType, Allocator>::rd_string(const char* con
       if (((pos = str.find('e')) != std::string::npos) || ((pos = str.find('E')) != std::string::npos))
       {
          // Remove the exponent part from the string.
+         #ifndef BOOST_MP_STANDALONE
          exp = boost::lexical_cast<exponent_type>(static_cast<const char*>(str.c_str() + (pos + 1u)));
+         #else
+         BOOST_IF_CONSTEXPR(std::is_integral<exponent_type>::value)
```

Comment:
The exponent type is definitely an integral type.

Reply:
Correct.

Reply:
Will the exponent always be a built-in integral type? That was this is trying to catch.

Reply:
Yes, we don't support anything else.  64-bit exponents support a crazy-large enough range anyway!